CRM/Utils/SQL.php

```diff
@@ -76,4 +76,24 @@ public static function supportsFullGroupBy() {
     return version_compare(CRM_Core_DAO::singleValueQuery('SELECT VERSION()'), '5.7', '>=');
   }
 
+  /**
+   * Disable ONLY_FULL_GROUP_BY for MySQL versions lower then 5.7
+   *
+   * @return bool
+   */
+  public static function disableFullGroupByMode() {
+    $sqlModes = self::getSqlModes();
+
+    // Disable only_full_group_by mode for lower sql versions.
+    if (!self::supportsFullGroupBy() || (!empty($sqlModes) && !in_array('ONLY_FULL_GROUP_BY', $sqlModes))) {
```

Comment:
@monishdeb should the 2nd half of that 2nd test be && in_array not && !in_array as we want it if ONLY_FULL_GROUP_BY is in the array right?

Reply:
@seamuslee001 no because the second part ```(!empty($sqlModes) && !in_array('ONLY_FULL_GROUP_BY', $sqlModes))``` is for Mysql 5.7 where if ONLY_FULL_GROUP_BY is intentionally not present (i.e. !in_array) in sql modes then return TRUE which means GroupByMode is disabled.

Reply:
so you expect then in would just jump to the return in L93 because the array_search in 89 won't find it then right?

Reply:
yes

Reply:
ok no worries makes sense then